aurora/core.py

```diff
@@ -888,8 +1805,119 @@ def run_aurora(
+                rcld_impl_rate,
+                rcld_sput_rate,
+                rclb_rate,
+                rcls_rate,
+                rclp_rate,
+                rclw_rate,
+                rclw_refl_rate,
+                rclw_recl_rate,
+                rclw_impl_rate,
+                rclw_sput_rate
+            ) = self.res
+        
+        # Return number of elements in the output of the simulation consistently with
+        #   the number of reservoirs/fluxes defined in the wall interaction / recycling model
+        #   in such a way that, if the new features of the advanced recycling model are
+        #   de-activated, only the "standard" fields are returned in the out tuple,
+        #   in the previous and correct order, and backwards compatibility is ensured
+        # The differences with respect to the versions before the introduction of the
+        #   advanced recycling model are highlighted
+        
+        out = (nz,) # same as before
+        
+        out = out + (N_mainwall,) # corresponding to N_wall in the old nomenclature
+        
+        if self.div_recomb_ratio < 1.0: # ions allowed to interact with a divertor wall
+            out = out + (N_divwall,) # new field
+        
+        out = out + (N_div,) # same as before
+        
+        if self.pump_chamber: # a second reservoirs for neutral particles before the pump is defined
+            out = out + (N_pump,) # new field with different meaning wrt the old N_pump
+        
+        out = out + (N_out,) # corresponding to N_pump in the old nomenclature
+        
+        out = out + (N_mainret,) # corresponding to N_ret in the old nomenclature
+        
+        if self.div_recomb_ratio < 1.0: # ions allowed to interact with a divertor wall
+            out = out + (N_divret,) # new field
+            
+        out = out + (N_tsu,) # same as before
+        
+        out = out + (N_dsu,) # same as before
+        
+        out = out + (N_dsul,) # same as before
+        
+        if self.div_recomb_ratio < 1.0: # ions allowed to interact with a divertor wall
+            out = out + (rcld_rate,) # new field with different meaning wrt the old rcld_rate
+            
+            if self.advanced_PWI_flag:
+                out = out + (rcld_refl_rate,rcld_recl_rate,rcld_impl_rate,rcld_sput_rate,)
+            
+        out = out + (rclb_rate,) # corresponding to rcld_rate in the old nomenclature
+
+        if self.screening_eff > 0.0: # the backflow from the divertor can be screened before reaching the core plasma
+            out = out + (rcls_rate,) # new field
+        
+        if self.pump_chamber: # a second reservoirs for neutral particles before the pump is defined
+            out = out + (rclp_rate,) # new field
+
+        out = out + (rclw_rate,) # same as before  
+                    
+        if self.advanced_PWI_flag:
+            out = out + (rclw_refl_rate,rclw_recl_rate,rclw_impl_rate,rclw_sput_rate,)
+
+        return out
```

Comment:
@odstrcilt as discussed privately with @fsciortino , I totally agree that the best form out should be as a dictionary. The reason why I tried to implement it in the code as it is now (although I find it quite not elegant by myself) was to maintain backwards compatibility at all costs, regarding the interaction of users with out. But unfortunately, since out was previously a tuple, any addition of fields will break this backwards compatibility, unless one does something very ugly as I did.
If we turn it into a dictionary, users will have to change their personal routines in the way in which they unpack out. But I think that this is a necessary sacrifice, so that adding, also in the future, fields to out will be much more simple, and users will be able to call jus the fields in which they are interested to as out["field_name"].